### DISCUSSION TITLE
How to write tests in salvo?

### DISCUSSION THREAD

**Discussion by TheAwiteb** [salvo-rs/salvo#631] [Q&A] [answered]
[I have a function that returns the API service], so I put it in the `RequestBuilder::send` function, but it's return 404, I don't know why.

Note: I send to `http://127.0.0.1:5801` with the my endpoint path. I see that salvo tests are sending to `http://127.0.0.1:5801` [here], so I think it's ok.

[here]: https://github.com/salvo-rs/salvo/blob/79ec3532033684765c2d342066b43e7bbbdaf4e6/crates/jwt-auth/src/lib.rs#L292
[I have a function that returns the API service]: https://github.com/TheAwiteb/xors/blob/5bb413cacb0c26099f097761c4e64072fb5ae3c8/xors-api/src/api/mod.rs#L90

**Comment by chrislearn** [comment]
You are test url "http://127.0.0.1:5801/hello", but there is no router to handle this path.

**Comment by TheAwiteb** [comment]
> but there is no router to handle this path.

I have test `http://127.0.0.1:5801/auth/captcha`, which the service have it

https://github.com/TheAwiteb/xors/blob/5bb413cacb0c26099f097761c4e64072fb5ae3c8/xors-api/src/api/mod.rs#L121-L126

**Reply by chrislearn** [reply]
?

**Reply by TheAwiteb** [reply, marked as answer]
Ohh, sorry it's was a bug in my router, the `captcha` should be GET but I make it POST (:
